### DISCUSSION TITLE
NextFont gives a type error on Build mode

### DISCUSSION THREAD

**Discussion by fadadOussama** [vercel/next.js#58693] [Help]
### Summary

When I run a build, I encounter the following TypeScript error :

```js
.next/types/app/layout.ts:8:13
Type error: Type 'OmitWithTag<typeof import("C:/Users/oussama/Desktop/LEARN/freelancer_dash/app/layout"), "metadata" | "default" | "config" | "generateStaticParams" | "revalidate" | "dynamic" | "dynamicParams" | ... 6 more ... | "generateViewport", "">' does not satisfy the constraint 
'{ [x: string]: never; }'.
  Property 'secondFont' is incompatible with index signature.
    Type 'NextFont' is not assignable to type 'never'.

   6 |
   7 | // Check that the entry is a valid entry
>  8 | checkFields<Diff<{
     |             ^
   9 |   default: Function
  10 |   config?: {}
  11 |   generateStaticParams?: Function
  ```
and when i visit the ".next/types/app/layout.ts" the same type error that in terminal is there as well :

```js
checkFields<Diff<{
  default: Function
  config?: {}
  generateStaticParams?: Function
  revalidate?: RevalidateRange<TEntry> | false
  dynamic?: 'auto' | 'force-dynamic' | 'error' | 'force-static'
  dynamicParams?: boolean
  fetchCache?: 'auto' | 'force-no-store' | 'only-no-store' | 'default-no-store' | 'default-cache' | 'only-cache' | 'force-cache'
  preferredRegion?: 'auto' | 'global' | 'home' | string | string[]
  runtime?: 'nodejs' | 'experimental-edge' | 'edge'
  maxDuration?: number
  
  metadata?: any
  generateMetadata?: Function
  viewport?: any
  generateViewport?: Function
  
}, TEntry, ''>>()
```
  This error seems to be related to my font declaration in my layout , specifically with the 'secondFont' variable, which is of type 'NextFont'. The TypeScript error says that there's an issue with the compatibility of types.

The font is declared in my main layout the like this :
```js
import { Playfair_Display } from "next/font/google";
export const secondFont = Playfair_Display({ subsets: ["latin"] });
```

While everything works fine in development mode, the build process raises this TypeScript error.

### Additional information

```js
next info :
Operating System:        
  Platform: win32        
  Arch: x64
  Version: Windows 10 Pro
Binaries:
  Node: 20.9.0
  npm: N/A
  Yarn: N/A
  pnpm: N/A
Relevant Packages:       
  next: 14.0.3
  eslint-config-next: 13.5.4
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.2.2
Next.js Config:
  output: N/A
```

Btw i tested in the latest version of canary ( v14.0.4-canary.5 ), but the error is still there.

**Comment by ikaushiksharma** [comment]
I had the same error. I was exporting twice from pages.tsx, first at the top as export async function Page(…) and again at the end of the file as export default Page;. The error was fixed after doing export default async function Page(…).
### So basically if you export a same component/object more than once in one or more files. you get this error

**Reply by CodexRB** [reply]
Thanks a lot 🙏🏽

**Comment by terrance456** [comment]
Removing export from font `const` seems to fix it

.e.g.


```tsx
export const fontSans: NextFontWithVariable = FontSans({
  subsets: ["latin"],
  variable: "--font-sans",
});

fix:
const fontSans: NextFontWithVariable = FontSans({
  subsets: ["latin"],
  variable: "--font-sans",
});
```

**Reply by sonyarianto** [reply]
ya me too, removing the export fix the problem.

context:
this is happen on my project that using shadcn/ui when following their docs

**Reply by nidalA1992** [reply]
thanks, its help me

**Reply by reliek21** [reply]
Thanks guys. That was helpful.

**Reply by mk48** [reply]
Yes, shadcn/ui is the reason 🤐

**Reply by manastm** [reply]
but what if we need to export the font to use elsewhere 😭

**Reply by seok-hyung** [reply]
thanks a lot

**Reply by dhruvsakariya** [reply]
Thanks a lot 👌
